### ISSUE TITLE
Set-and-forget suggestion?

### ISSUE THREAD

**Issue by yaronguez** [eealeivan/mixpanel-csharp#2]
Thanks so much for the great wrapper! Really well done. Can you recommend an approach to fire tracking messages in a set-and-forget fashion that doesn't hold up the application while awaiting for the async calls to complete? We're considering items described here: http://www.hanselman.com/blog/HowToRunBackgroundTasksInASPNET.aspx

**Comment by eealeivan** [comment]
Hi, do you use a web or windows app?

**Comment by yaronguez** [comment]
This is within an ASP.NET WebAPI project. We're planning on
using QueueBackgroundWorkItem as described here
https://blogs.msdn.microsoft.com/webdev/2014/06/04/queuebackgroundworkitem-to-reliably-schedule-and-run-background-processes-in-asp-net/


Unless you have a better approach...

Yaron Guez
Software Architect
(781) 249-4709

**Comment by eealeivan** [comment]
I think this approach might work.
I would suggest to first generate needed messages using `Get{method-type}Message` methods. For example:
```
MixpanelMessage msg1 = mc.GetTrackMessage("Bonus Gained", new 
{ 
    DistinctId = "12345",
    BonusName = "Booster"
});

MixpanelMessage msg2 = mc.GetTrackMessage("Level Up", new 
{ 
    DistinctId = "12345",
    Level = 3
});
```
And then adding them to background worker queue:
`HostingEnvironment.QueueBackgroundWorkItem(ct => SendToMixpanelAsync(msg1, msg2));`
In `SendToMixpanelAsync` method you create a `MixpanelClient` without parameters, because token will be set when messages were created.
```
 private async Task SendToMixpanelAsync(params MixpanelMessage[] messages)
 {
    var mc = new MixpanelClient();
    await mc.SendAsync(messages);
}
```
So, to summarize: in you actions you will just create messages and actual sending will be performed in background worker.

**Comment by yaronguez** [comment]
Looks good! We chose a similar approach:
```
public static void SaveChanges(List<MixpanelMessage> messages)
{
    HostingEnvironment.QueueBackgroundWorkItem(async ct =>
    {
        var mc = new MixpanelClient();
        await mc.SendAsync(messages);
    });
}
```
Thanks for your help and prompt reply. It's a great library!
